spec/models/course_user_spec.rb

```diff
@@ -164,6 +164,47 @@
       end
     end
 
+    describe 'soft delete behavior' do
```

Comment:
Your tests does not cover checks that associated models are deleted and restored?

Reply:
Ok, it should have